clmm/clusterensemble.py

```diff
@@ -113,38 +110,69 @@ def make_individual_radial_profile(self, galaxycluster, bin_units, bins=10, erro
-            include_empty_bins=True, return_binnumber=False,
-            cosmo=cosmo, z_lens=galaxycluster.z, weights=galaxycluster.galcat[weights_in],
-            components_error=[None if n is None else galaxycluster.galcat[n].data
-                              for n in (tan_component_in_err, cross_component_in_err, None)],
-            )
-        #profile_table[weights_out] = 1 #rm this line
-        if bin_units != 'radians':
-            bins = convert_units(bins, bin_units, 'radians', redshift=galaxycluster.z, cosmo=cosmo)
-        profile_table[weights_out] = binned_statistic(galaxycluster.galcat['theta'], 
-                                                       galaxycluster.galcat[weights_in], 
-                                                       statistic='sum', 
-                                                       bins = bins)[0]
-        data_to_save = [galaxycluster.unique_id, galaxycluster.ra, galaxycluster.dec, galaxycluster.z,
-                        *[np.array(profile_table[col]) for col in
-                            ('radius', 'p_0', 'p_1', weights_out)]]
-        # to be fixed down to here after issue 443 is merged
+        tb_kwargs = {}
+        tb_kwargs.update(locals())
+        tb_kwargs.pop('self')
+        tb_kwargs.pop('tb_kwargs')
+        tb_kwargs.pop('galaxycluster')
+
+        cl_bin_units = galaxycluster.galcat.meta.get('bin_units', None)
+        self.data.update_info_ext_valid('bin_units', self.data, cl_bin_units, overwrite=False)
+
+        cl_cosmo = galaxycluster.galcat.meta.get('cosmo', None)
+        self.data.update_info_ext_valid('cosmo', self.data, cl_cosmo, overwrite=False)
+
+        profile_table = galaxycluster.make_radial_profile(
+            include_empty_bins=True, gal_ids_in_bins=False, add=False,
+            **tb_kwargs)
+
+        self.add_individual_radial_profile(
+            galaxycluster, profile_table,
+            tan_component_out, cross_component_out, weights_out)
+
+    def add_individual_radial_profile(self, galaxycluster, profile_table,
+                                      tan_component='gt', cross_component='gx',
+                                      weights='W_l'):
+        """Compute the individual shear profile from a single GalaxyCluster object
+        and adds the averaged data in the data attribute.
+
+        Parameters
+        ----------
+        galaxycluster : GalaxyCluster
+            GalaxyCluster object with cluster metadata and background galaxy data
+        tan_component: string, optional
+            Name of the tangetial component binned column to be added in profile table.
+            Default: 'gt'
+        cross_component: string, optional
+            Name of the cross component binned profile column to be added in profile table.
+            Default: 'gx'
+        weights : str, None
+            Name of the weight column to be used in the added to the profile table.
+        """
```

Comment:
- [ ] `profile_table` is missing from the docstrings
- [ ] Rephrase decription the the `weights` keyword. Currently there's a problem in the sentence